src/Tests/dotnet.Tests/CommandTests/ToolRunCommandTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.CommandFactory;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.DotNet.ToolManifest;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.DotNet.Tools.Tool.Run;
+using Microsoft.Extensions.DependencyModel.Tests;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Tests.Commands.Tool
+{
+    public class ToolRunCommandTests 
+    {
+        private readonly IFileSystem _fileSystem;
+        private const string ManifestFilename = "dotnet-tools.json";
+        private readonly string _testDirectoryRoot;
+        private DirectoryPath _nugetGlobalPackagesFolder;
+        private readonly LocalToolsResolverCache _localToolsResolverCache;
+
+        public ToolRunCommandTests() 
+        {
+            _fileSystem = new FileSystemMockBuilder().UseCurrentSystemTemporaryDirectory().Build();
```

Comment:
When writing new tests, I think we should prefer deriving from `SdkTest` and using `_testAssetsManager.CreateTestDirectory` instead of using the temporary directory.  This means we won't be able to set up as much in the constructor (we can use a helper method instead), but makes it easier to debug / investigate tests, since they use a predictable directory that's easier to find.

Reply:
Thanks for the suggestion